### PR TITLE
SRM: VictorOps change source

### PR DESCRIPTION
**Pull request by SudheendraKatte**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Comment by SudheendraKatte** [comment]
@deepak-harness @shankaranandh Please review and provide comments.

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+description: Configure VictorOps change source to monitor incidents.
+sidebar_position: 70
+---
+
+# VictorOps change source
+
+You can configure VictorOps as a custom change source to monitor incidents.
+
+
+Configuring VictorOps as a change source involves the following steps:
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:12]:
```suggestion
To configure VictorOps as a change source, do the following:
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+1. [Add VictorOps as a custom change source in the Harness SRM monitored service and generate a webhook.](#add-victorops-as-a-change-source)
+   
+[Integrate Harness webhook with VictorOps.](#integrate-harness-webhook-with-victorops)
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:16]:
```suggestion
2. [Integrate Harness webhook with VictorOps.](#integrate-harness-webhook-with-victorops)
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+---
+title: VictorOps
+description: Configure VictorOps change source to monitor incidents.
+sidebar_position: 70
+---
+
+# VictorOps change source
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:7]:
```suggestion
# Configure VictorOps change source
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+Configuring VictorOps as a change source involves the following steps:
+
+1. [Add VictorOps as a custom change source in the Harness SRM monitored service and generate a webhook.](#add-victorops-as-a-change-source)
+   
+[Integrate Harness webhook with VictorOps.](#integrate-harness-webhook-with-victorops)
+
+
+## Add VictorOps as a change source
+
+To add VictorOps as a change source to a monitored service:
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:21]:
```suggestion
To add VictorOps as a change source to a monitored service, do the following:
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    The custom incident change source gets added to the monitored service. A webhook URL and cURL commands are generated.
+
+    <docimage path={require('./static/add-victorops-change-source.png')} />
+
+
+## Copy webhook URL
+
+To configure your VictorOps to send incident reports to the monitored service in Harness SRM, you'll need to obtain a webhook from the SRM monitored service.
+
+To copy the webhook URL from the monitored service:
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:45]:
```suggestion
To copy the webhook URL from the monitored service, do the following:
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+2. On the **Configurations** page, go to the **Change Sources** tab. In the **Change Sources** section, a list of change sources that have been added to the monitored service is displayed. The list also displays webhook URLs and cURL commands next to each change source.
+
+3. Locate the VictorOps incident change source that you want to configure with the VictorOps, and then copy the webhook URL.
+
+
+## Integrate Harness webhook with VictorOps
+
+You need to integrate the webhook you created in Harness SRM with VictorOps to start receiving the incidents alerts.
+
+To integrate Harness webook with VictorOps:
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:58]:
```suggestion
To integrate Harness webook with VictorOps, do the following:
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+
+## Add VictorOps as a change source
+
+To add VictorOps as a change source to a monitored service:
+
+1. In your Harness project, go to **Service Reliability** > **Monitored Services**.  
+   A list of monitored services is displayed.
+
+2. Locate the monitored service for which you want to add a custom change source, select the three vertical dots next to it, and then select **Edit service**.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:26]:
```suggestion
2. Locate the monitored service for which you want to add a custom change source, select **More Options** (&vellip;), and then select **Edit service**.  
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+   The Configurations page appears.
+
+3. On the **Configurations** page, go to the **Change Sources**, and select **+ Add New Change Source**.  
+
+4. On the Edit Change Source page, under **Select Change Source**, choose **Incident** as **Provider Type**, and then select **Custom**.
+
+5. Enter a name for the change source. For example, VictoreOps.
+
+6.  Select **Submit**.  
+    The custom incident change source gets added to the monitored service. A webhook URL and cURL commands are generated.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:36]:
```suggestion
    The custom incident change source is added to the monitored service. A webhook URL and cURL commands are generated.
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+To integrate Harness webook with VictorOps:
+
+1. Log in to VictorOps and select **Integrations** on the menu bar.
+
+2. On the Integrations page, go to the **Outgoing Webhooks** tab.
+
+3. Select **Add Webhook** and configure the following settings:
+   
+   - **Event**: Choose **Incident-triggered** to specify when the webhook should trigger.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:66]:
```suggestion
   - **Event:** Choose **Incident-triggered** to specify when the webhook should trigger.
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+   - **Method**: Select the **POST** method.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:67]:
```suggestion
   - **Method:** Select the **POST** method.
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+   - **Content-Type**: Set this to **application/json**.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:68]:
```suggestion
   - **Content-Type:** Set this to **application/json**.
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+   - **Custom-Header**: Include an X-api-Key header for authentication and security. You can create this key in your VictorOps account.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:69]:
```suggestion
   - **Custom-Header:** Include an X-api-Key header for authentication and security. You can create this key in your VictorOps account.
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+   - **To**: Paste the webhook URL copied during the custom change source creation in Harness SRM.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:70]:
```suggestion
   - **To:** Paste the webhook URL copied during the custom change source creation in Harness SRM.
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+   - **Payload**: Define the payload structure as follows:
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:71]:
```suggestion
   - **Payload:** Define the payload structure as follows:
```

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+      }
+
+      ```
+
+4.  Select **Save** to complete the integration setup.
+
+
+### Advanced Option: Customizing Webhooks for Different Monitored Services
+
+If you need to trigger different webhooks depending on specific rules or want to associate different webhook URLs with each custom change source in various monitored services, you can achieve this through VictorOps Alert Rules Engine.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:92]:
```suggestion
If you need to trigger different webhooks depending on specific rules or want to associate different webhook URLs with each custom change source in various monitored services, you can do this through the VictorOps Alert Rules Engine.
```

**Review by brian-f-harness — DISMISSED** [review]
Looks good. Left a few minor suggestions.

File: docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md (hunk 9 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/change-source/victorops-change-source.md:66]: Retaining the current style

**Comment by shankaranandh** [comment]
looks good.

**Pull request merged**